### PR TITLE
Feature/local files

### PR DESCRIPTION
**Pull request by klaseskilson**
This PR features:
- Local identity added to user if the user `own_password`
- Local files downlodable

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -13,8 +13,11 @@ class User < ActiveRecord::Base
                     on: :create
   validates_presence_of :name, :on => :create
 
-  # before actions
+  # before/after hooks
   before_destroy :delete_identities
+  after_create :send_welcome_email
```
Comment (danielronnkvist) [inline app/models/user.rb]: This is deprecated.

File: app/models/user.rb [hunk 2]
```diff
@@ -14,8 +14,10 @@ class User < ActiveRecord::Base
                     on: :create
   validates_presence_of :name, :on => :create
 
-  # before actions
+  # before/after hooks
   before_destroy :delete_identities
+  after_create :append_local_identity
+  after_update :append_local_identity
```
Comment (danielronnkvist) [inline app/models/user.rb]: These could be put on the same row with `after_save :append_local_identity`.

File: app/models/user.rb [hunk 3]
```diff
@@ -66,4 +68,19 @@ def delete_identities
...
+    # Private: Create an identity for local file handling when user is created.
+    # Future: This should only happen if user has a password.
+    #
+    # Returns nothing
+    def append_local_identity
+      if own_password && !identities.exists?(service: 'local')
+        local_identity = Identity.new(service: 'local',
+                                      token: SecureRandom.hex(5),
+                                      name: 'Bruse',
+                                      uid: SecureRandom.hex(5))
```
Comment (danielronnkvist) [inline app/models/user.rb]: What happens if _uid_ is equal to some other _uid_ and an error is raised? Perhaps a check to see that the `local_identity` really is created.

**Comment by danielronnkvist** [comment]
Looks good. I only had two comments, when they have been taken care of I say go! :rocket:

**Comment by danielronnkvist** [comment]
One more thing, should the home view be updated as well to not display local identity?

File: app/models/user.rb (hunk 3 above)
Comment (klaseskilson) [inline app/models/user.rb]: I'm changing the uid to use `SecureRandom.uuid` instead.

From [Wikipedia](http://en.wikipedia.org/wiki/Universally_unique_identifier):

> The identifier size and generation process need to be selected so as to make this sufficiently improbable in practice. Anyone can create a UUID and use it to identify something with reasonable confidence that the same identifier will never be unintentionally created by anyone to identify something else. Information labeled with UUIDs can therefore be later combined into a single database _without needing to resolve identifier_ (ID) conflicts.

File: app/models/user.rb (hunk 2 above)
Comment (tkomstadius) [inline app/models/user.rb]: Where does the save take place in relation to create and update?

**Comment by tkomstadius** [comment]
I don't really understand the last comment. hehe... But otherwise I think it looks pretty. I've asked a lot of questions, got a lot of answers. Looks mergable to me.

**Comment by danielronnkvist** [comment]
Looks fine! Merge if you want @tkomstadius :+1:

**Comment by tkomstadius** [comment]
I do

**Pull request merged**